src/librustdoc/passes/mod.rs

```diff
@@ -1,122 +1,29 @@
-//! Contains information about "passes", used to modify crate information during the documentation
-//! process.
-
-use self::Condition::*;
-use crate::clean;
-use crate::core::DocContext;
+//! The definitions of *passes* which transform crate information.
 
 mod stripper;
 pub(crate) use stripper::*;
 
-mod strip_aliased_non_local;
-pub(crate) use self::strip_aliased_non_local::STRIP_ALIASED_NON_LOCAL;
-
-mod strip_hidden;
-pub(crate) use self::strip_hidden::STRIP_HIDDEN;
-
-mod strip_private;
-pub(crate) use self::strip_private::STRIP_PRIVATE;
-
-mod strip_priv_imports;
-pub(crate) use self::strip_priv_imports::STRIP_PRIV_IMPORTS;
-
-mod propagate_doc_cfg;
-pub(crate) use self::propagate_doc_cfg::PROPAGATE_DOC_CFG;
-
-mod propagate_stability;
-pub(crate) use self::propagate_stability::PROPAGATE_STABILITY;
-
+pub(crate) mod calculate_doc_coverage;
+pub(crate) mod check_doc_test_visibility;
 pub(crate) mod collect_intra_doc_links;
-pub(crate) use self::collect_intra_doc_links::COLLECT_INTRA_DOC_LINKS;
-
-mod check_doc_test_visibility;
-pub(crate) use self::check_doc_test_visibility::CHECK_DOC_TEST_VISIBILITY;
-
-mod collect_trait_impls;
-pub(crate) use self::collect_trait_impls::COLLECT_TRAIT_IMPLS;
-
-mod calculate_doc_coverage;
-pub(crate) use self::calculate_doc_coverage::CALCULATE_DOC_COVERAGE;
-
-mod lint;
-pub(crate) use self::lint::RUN_LINTS;
-
-/// A single pass over the cleaned documentation.
-///
-/// Runs in the compiler context, so it has access to types and traits and the like.
-#[derive(Copy, Clone)]
-pub(crate) struct Pass {
-    pub(crate) name: &'static str,
-    pub(crate) run: Option<fn(clean::Crate, &mut DocContext<'_>) -> clean::Crate>,
-    pub(crate) description: &'static str,
+pub(crate) mod collect_trait_impls;
+pub(crate) mod lint;
+pub(crate) mod propagate_doc_cfg;
+pub(crate) mod propagate_stability;
+pub(crate) mod strip_aliased_non_local;
+pub(crate) mod strip_hidden;
+pub(crate) mod strip_priv_imports;
+pub(crate) mod strip_private;
+
+macro_rules! initialize {
```

Comment:
I'm not sure how you feel about these macro shenanigans (`track` & `initialize`). I'm split.

`track` only exists so we can don't need to manually name each pass for the self profiler. On the other hand it's a bit DSL-y. However, I definitely don't want to go back to having an extra `const` item per pass for the "metadata".